### ISSUE TITLE
Mnimal example cannot work

### ISSUE THREAD

**Issue by Kami-chanw** [Daguerreo/QMLTreeView#13]
When adds the first new item here
```cpp
// main.cpp
void populateModel(TreeModel& model)
{
   auto america = new TreeItem("America");
   auto asia = new TreeItem("Asia");
   auto europe = new TreeItem("Europe");
   auto oceania = new TreeItem("Oceania");

   auto brazil = new TreeItem("Brazil");
   auto canada = new TreeItem("Canada");
   auto mexico = new TreeItem("Mexico");
   auto usa = new TreeItem("USA");

   auto italy = new TreeItem("Italy");
   auto france = new TreeItem("France");
   auto portugal = new TreeItem("Portugal");
   auto spain = new TreeItem("Spain");

   model.addTopLevelItem(america); // <--- assertion failed here
...
```
an assertion failure will be triggered due to `parent.childrenCount() - 1 == -1`
```cpp
// tree_model.h
void TreeModel::addItem(TreeItem* parent, TreeItem* child) {
    if (!child || !parent) {
        return;
    }

    emit layoutAboutToBeChanged();

    if (child->parentItem()) {
        beginRemoveRows(QModelIndex(), child->parentItem()->childCount() - 1, child->parentItem()->childCount());
        child->parentItem()->removeChild(child);
        endRemoveRows();
    }

    beginInsertRows(QModelIndex(), parent->childCount() - 1, parent->childCount() - 1); // bug
    child->setParentItem(parent);
    parent->appendChild(child);
    endInsertRows();

    emit layoutChanged();
}
```

**Comment by guorenwei321** [comment]
Can anyone fix this bugs?   How did you fix?

**Comment by Daguerreo** [comment]
Sorry, I completely missed this one.

May I have some additional information? Which Qt version are you using? Which OS?

This plugin is meant for Qt 5.15 which did not provide a TreeView. Since Qt 6.3 TreeView has been added and the properties are almost the same of my version, so for a quick fix I would suggest to use the standard one.

I'm experiencing the crash with Qt 6.5, I'm going to look at it when I can.

**Comment by Kami-chanw** [comment]
I'm using Qt 6.5 on WIndows.
It is easy to fix, you just need to replace the line I marked bug with the following line
```
beginInsertRows(QModelIndex(), qMax(parent->childCount() - 1, 0), qMax(parent->childCount() - 1, 0)); 
```

Besides, when I subclass `QSortFilterProxyModel` and `setSource(treeModel)`, it will crash when `emit layoutChanged()` in `addItem`. And what I have done is overriding the `lessThan` from `QSortFilterProxyModel`. I'm not sure whether it is your problem.

**Comment by Daguerreo** [comment]
Then please open a MR to address the issue

**Comment by Daguerreo** [comment]
Fixed and released v1.0.0
